lib/modules/manager/npm/post-update/rules.ts

```diff
@@ -19,8 +19,13 @@ export function processHostRules(): HostRulesResult {
   const npmHostRules = hostRules.findAll({
     hostType: 'npm',
   });
-  logger.debug(`Found ${npmHostRules.length} npm host rule(s)`);
-  for (const hostRule of npmHostRules) {
+  // Include host rules without specific type to mimic the behavior used when determining dependencies with updates.
+  const noTypeHostRules = hostRules
+    .getAll()
+    .filter((rule) => rule.hostType === null || rule.hostType === undefined);
+  const effectiveHostRules = npmHostRules.concat(noTypeHostRules);
+  logger.debug(`Found ${effectiveHostRules.length} npm host rule(s)`);
+  for (const hostRule of effectiveHostRules) {
```

Comment:
You need to dedupe first. Ones with hostType=npm should have higher priority

Reply:
Added. The implementation currently assumes that the list of rules is not terribly long. Otherwise, a more efficient data structure should be used for deduplication. Is that fine?

Reply:
It shouldn't be particularly long